src/tabs.ts

```diff
@@ -0,0 +1,307 @@
+            if(val != this._internal_fashion)
+                this._changeFashionStyle(val);
+        } else if(name == "colsize") {
+            this._column_size = parseAttrVal(new_val, (v:string) => {
+                try {
+                    CSSStyleValue.parse("grid-template-columns", v)
+                    return true;
+                } catch (err:any) {}
+                    return false;
+            });
+        }
+    }
+    override render() {
+        if(this._internal_fashion == 'accordion') {
+            return html`
+<div @click=${this._titleClicked}
+     @animationend=${ (event:any) => this._animationEndEvent(event)}>
+    <slot name='acc'></slot>
+</div>`
+        } else {
+            let fashionColumn:any = {
+                display: 'grid',
+                'grid-template-columns': `
+                    ${this._column_size.title} ${this._column_size.body}`
+            }
+            return html`
+${this._animateStyles()}
+<div style=${this._internal_fashion=='column'?
+                styleMap(fashionColumn):''}>
+    <div id="title-tab" style=
+        "display:flex;
+         flex-direction:${this._internal_fashion}">
+        <slot name="tab-title" @click=${this._titleClicked}></slot>
+    </div>
+    <div id="body-tab" @animationend=${ (event:any) => this._animationEndEvent(event)}>
+        <slot id="body-slot" name="tab-body"></slot>
+    </div>
+</div>`
+        }
+    }
+    override connectedCallback() {
+        super.connectedCallback();
+        this.renderRoot.addEventListener("slotchange", (e) => {
+            this._slotChangedEvent(e)});
+        window.addEventListener("resize", this._resizeListener);
+        this.addEventListener("DOMNodeInserted", this._node_insert_listener);
+    }
+    override disconnectedCallback() {
+        window.removeEventListener("resize", this._resizeListener);
+        this.removeEventListener("DOMNodeInserted", this._node_insert_listener);
+    }
+    override firstUpdated() {
+        if(!this.animation || (
+                !this._animation_styles.open && !this._animation_styles.close))
+            this.animationDur = 0;
+        this._openActive();
+        if(this._cur_ix == null)
+            this.show(0);
+    }
+    private _nodeInserted(event:Event) {
+        let el = event.target as HTMLElement;
+        if (this._internal_fashion != 'accordion')
+            return;
+        if(el.matches("[slot=tab-title]") || el.matches("[slot=tab-body]"))
+            this._showAccordion();
+        else
+            return;
+    }
+    private _slotChangedEvent(event:Event) {
+        if(!event.target)
+            return;
+        let nodes = (event.target as HTMLSlotElement).assignedElements();
+        let name = (event.target as HTMLSlotElement)?.name;
+        if(name == 'tab-body') {
+            for(let i of nodes)
+                if(!i.classList.contains(this.activecls))
+                    (i as HTMLElement).style.display = 'none';
```

Comment:
When a new tab with active class is added , we would end showing mutiple tab elements body.